run_unittests.py

```diff
@@ -30,6 +30,8 @@ def get_soname(fname):
         m = pattern.search(line)
         if m is not None:
             return m.group(1)
+    raise RuntimeError('Could not determine soname:\n\n' +
+                       raw_out.decode('utf8', errors='ignore'))
```

Comment:
This will conflict with https://github.com/mesonbuild/meson/pull/1108/commits/59bf3ab98fcb06d35d3f66564b6c81dd22531d32. You don't need to do any `decode()` once you use `universal_newlines`.